api/src/OpenTelemetry/Internal/Common/Types.hs

```diff
@@ -4,7 +4,13 @@
 module OpenTelemetry.Internal.Common.Types (
   InstrumentationLibrary (..),
   ShutdownResult (..),
+  takeWorseShutdownResult,
+  takeWorstShutdownResult,
+  flushResultToShutdownResult,
   FlushResult (..),
+  takeWorseFlushResult,
+  takeWorstFlushResult,
+  exportResultToFlushResult,
```

Comment:
Helper functions to convert from `ExportResult -> FlushResult` and `FlushResult -> ShutdownResult`. These are used to propagate Failures and Timeouts up the chain of `export` to `forceFlush` to `shutdown`.